spec/unit/actions/synchronize_user_filters_spec.rb

```diff
@@ -74,4 +54,59 @@
       end
     end
   end
+
+  context 'when using sync_one_project_based_on_custom_id mode with multiple_projects_column' do
+    let(:params) do
+      params = {
+        GDC_GD_CLIENT: client,
+        input_source: 'foo',
+        domain: 'bar',
+        filters_config: { labels: [] },
+        multiple_projects_column: 'id_column',
+        sync_mode: 'sync_one_project_based_on_custom_id'
+      }
+      GoodData::LCM2.convert_to_smart_hash(params)
+    end
+
+    context 'when params do not match client data in domain' do
+      before do
+        allow(project).to receive(:metadata).and_return({})
+        allow(project).to receive(:uri).and_return('project-uri')
+        allow(domain).to receive(:clients).and_return([])
+      end
+
+      it 'fails when unable to get filter value for selecting filters' do
+        expect { subject.class.call(params) }.to raise_exception
```

Comment:
Please specify _which_ exception. Otherwise this can easily pass when an unexpected exception is raised. E.g. `raise_exception(/regex to match/)`